presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -207,16 +225,39 @@ public void testSTCentroid()
         assertCentroid("POLYGON ((1 1, 5 1, 3 4))", new Point(3, 2));
         assertCentroid("MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((2 4, 2 6, 6 6, 6 4)))", new Point(3.3333333333333335, 4));
         assertCentroid("POLYGON ((0 0, 0 5, 5 5, 5 0, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))", new Point(2.5416666666666665, 2.5416666666666665));
+    }
 
+    @Test
+    public void testSTCentroidInvalidGeometry()
+    {
         // invalid geometry
-        assertInvalidFunction("ST_Centroid(ST_GeometryFromText('MULTIPOLYGON (((4.903234300000006 52.08474289999999, 4.903234265193165 52.084742934806826, 4.903234299999999 52.08474289999999, 4.903234300000006 52.08474289999999)))'))", "Cannot compute centroid: .* Use ST_IsValid to confirm that input geometry is valid or compute centroid for a bounding box using ST_Envelope.");
+        assertApproximateCentroid("MULTIPOLYGON (((4.903234300000006 52.08474289999999, 4.903234265193165 52.084742934806826, 4.903234299999999 52.08474289999999, 4.903234300000006 52.08474289999999)))", new Point(4.9032343, 52.0847429), 1e-7);
+    }
+
+    @Test
+    public void testSTCentroidNumericStability()
+    {
+        String wkt1 = "MULTIPOLYGON (((153.492818 -28.13729, 153.492821 -28.137291, 153.492816 -28.137289, 153.492818 -28.13729)))";
+        assertApproximateCentroid(wkt1, new Point(153.49282, -28.13729), 1e-5);
+
+        String wkt2 = "MULTIPOLYGON (((153.112475 -28.360526, 153.1124759 -28.360527, 153.1124759 -28.360526, 153.112475 -28.360526)))";
+        assertApproximateCentroid(wkt2, new Point(153.112475, -28.360526), 1e-5);
     }
 
     private void assertCentroid(String wkt, Point centroid)
     {
         assertFunction(format("ST_AsText(ST_Centroid(ST_GeometryFromText('%s')))", wkt), VARCHAR, new OGCPoint(centroid, null).asText());
     }
 
+    private void assertApproximateCentroid(String wkt, Point centroid, double epsilon)
+    {
+        OGCGeometry geometry = OGCGeometry.fromText(wkt);
+        Slice inputSlice = GeometrySerde.serialize(geometry);
+        OGCPoint calculatedCentroid = (OGCPoint) GeometrySerde.deserialize(stCentroid(inputSlice));
+        assertEquals(calculatedCentroid.X(), centroid.getX(), epsilon);
```

Comment:
I'd rename centroid to expectedCentroid and calculatedCentroid to centroid (or actualCentroid).